invenio/modules/relationships/engines/sqlalchemy.py

```diff
@@ -0,0 +1,117 @@
+from uuid import uuid4
+
+from sqlalchemy_utils import UUIDType
+
+from invenio.ext.sqlalchemy import db, utils
+
+
+class SQLAlchemyNode(object):
+
+    """TODO"""
+
+    def _edges(self, inwards=True, outwards=True, link_type=None):
+        """TODO"""
+        query = SQLAlchemyEdge.query
+
+        if inwards and outwards:
+            query = query.filter((SQLAlchemyEdge.id_to == self['_id']) |
+                                 (SQLAlchemyEdge.id_from == self['_id']))
```

Comment:
Well, the `link_type` happen to be unique, but is not necessary the case: imagine a link type: "`childrenof`". This is valid no matter of the categories involved.

Additionally it is optional in the signature of your function ;-)